context/context.go

```diff
@@ -0,0 +1,76 @@
+package context
+
+import (
+	gocontext "context"
+	"fmt"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type Context interface {
+	gocontext.Context
+	WithGoContext(c gocontext.Context) Context
+	Errorf(err error)
+	StartSpan(tracer trace.Tracer, spanName string) (Context, trace.Span)
```

Comment:
This is returning a Context that isn't useful to other contexts embedding it.

Reply:
What do you mean ? useful as in ?

Reply:
If incident commander calls `StartSpan` it is returned `commons.Context` whereas it needs its own `api.Context`. It might have to always do type assertion